src/geo/lng_lat.js

```diff
@@ -98,30 +101,33 @@ class LngLat {
     toString() {
         return `LngLat(${this.lng}, ${this.lat})`;
     }
-}
 
-/**
- * Converts an array of two numbers to a `LngLat` object.
- *
- * If a `LngLat` object is passed in, the function returns it unchanged.
- *
- * @param {LngLatLike} input An array of two numbers to convert, or a `LngLat` object to return.
- * @returns {LngLat} A new `LngLat` object, if a conversion occurred, or the original `LngLat` object.
- * @example
- * var arr = [-73.9749, 40.7736];
- * var ll = mapboxgl.LngLat.convert(arr);
- * ll;   // = LngLat {lng: -73.9749, lat: 40.7736}
- */
-LngLat.convert = function (input) {
-    if (input instanceof LngLat) {
-        return input;
-    } else if (input && input.hasOwnProperty('lng') && input.hasOwnProperty('lat')) {
-        return new LngLat(input.lng, input.lat);
-    } else if (Array.isArray(input) && input.length === 2) {
-        return new LngLat(input[0], input[1]);
-    } else {
+    /**
+     * Converts an array of two numbers to a `LngLat` object.
+     *
+     * If a `LngLat` object is passed in, the function returns it unchanged.
+     *
+     * @param {LngLatLike} input An array of two numbers to convert, or a `LngLat` object to return.
+     * @returns {LngLat} A new `LngLat` object, if a conversion occurred, or the original `LngLat` object.
+     * @example
+     * var arr = [-73.9749, 40.7736];
+     * var ll = mapboxgl.LngLat.convert(arr);
+     * ll;   // = LngLat {lng: -73.9749, lat: 40.7736}
+     */
+    static convert(input: mixed): LngLat {
+        if (input instanceof LngLat) {
+            return input;
+        }
+        if (Array.isArray(input) && input.length === 2) {
+            return new LngLat(Number(input[0]), Number(input[1]));
+        }
+        if (!Array.isArray(input) && typeof input === 'object' && input !== null) {
+            return new LngLat(Number(input.lng), Number(input.lat));
+        }
         throw new Error("`LngLatLike` argument must be specified as a LngLat instance, an object {lng: <lng>, lat: <lat>}, or an array of [<lng>, <lat>]");
     }
-};
+}
+
+/*:: export type LngLatLike = LngLat | {lng: number, lat: number} | [number, number]; */
```

Comment:
@mourner unfortunately, no. eslint throws a parse error that's not ignorable because `sourceType` is set to "script" in [.eslintrc](https://github.com/mapbox/mapbox-gl-js/blob/master/.eslintrc). On top of this, Flow doesn't support requiring types using CommonJS. There may be a way to configure eslint to ignore type imports, but I haven't found it. An alternative is to migrate all the types to comments. Otherwise, I could see it being annoying if only some types are commented and others aren't.